### PR TITLE
www-client/qutebrowser: Add version bump 1.8.1

### PR DESCRIPTION
**Pull request by GuillaumeSeren**
Closes: https://bugs.gentoo.org/700372
Signed-off-by: Guillaume Seren guillaumeseren@gmail.com
Package-Manager: Portage-2.3.76, Repoman-2.3.16

### CODE REVIEW

File: dev-python/pypeg2/pypeg2-2.15.2.ebuild [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # Copyright 1999-2019 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=5
-PYTHON_COMPAT=( python{2_7,3_5,3_6} )
+EAPI=7
```
Comment (juippis) [inline dev-python/pypeg2/pypeg2-2.15.2.ebuild]: Bumping EAPI requires a revbump to the ebuild as well. So I'd suggest just updating PYTHON_COMPAT for now.

File: www-client/qutebrowser/qutebrowser-1.8.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+fi
+
+DESCRIPTION="A keyboard-driven, vim-like browser based on PyQt5 and QtWebEngine"
+HOMEPAGE="https://www.qutebrowser.org/ https://github.com/qutebrowser/qutebrowser"
+
+LICENSE="GPL-3"
+SLOT="0"
+IUSE="scripts test"
+
+COMMON_DEPEND="dev-python/setuptools[${PYTHON_USEDEP}]"
```
Comment (juippis) [inline www-client/qutebrowser/qutebrowser-1.8.1.ebuild]: I doubt this is true, because setuptools should be BDEPEND and not DEPEND. I accept it can go to RDEPEND, but I'll need to compile and check if that is true either.

File: www-client/qutebrowser/qutebrowser-1.8.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,87 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+PYTHON_COMPAT=( python{3_5,3_6,3_7} )
```
Comment (juippis) [inline www-client/qutebrowser/qutebrowser-1.8.1.ebuild]: One empty line below EAPI please.

**Comment by juippis** [comment]
Also please rebase against fresh ::gentoo and edit Github Issue title, appending `[please reassign]` there.

File: dev-python/pypeg2/pypeg2-2.15.2.ebuild (hunk 1 above)
Comment (GuillaumeSeren) [reply dev-python/pypeg2/pypeg2-2.15.2.ebuild]: Ok I rollback to EAPI=5

File: www-client/qutebrowser/qutebrowser-1.8.1.ebuild (hunk 2 above)
Comment (GuillaumeSeren) [reply www-client/qutebrowser/qutebrowser-1.8.1.ebuild]: Aggree I will change it to BDEPEND

File: www-client/qutebrowser/qutebrowser-1.8.1.ebuild (hunk 3 above)
Comment (GuillaumeSeren) [reply www-client/qutebrowser/qutebrowser-1.8.1.ebuild]: OK

File: dev-python/pypeg2/pypeg2-2.15.2.ebuild (hunk 1 above)
Comment (juippis) [reply dev-python/pypeg2/pypeg2-2.15.2.ebuild]: You have it the wrong way now :P your qutebrowser is EAPI-5 while this remains EAPI-7. Also remember that there is no BDEPEND in EAPI-5, so just update PYTHON_COMPAT.

I can bump this PYTHON_COMPAT and then just merge your qutebrowser contribution if you find rebasing hard/annoying.

Comment (GuillaumeSeren) [reply dev-python/pypeg2/pypeg2-2.15.2.ebuild]: Thank you @juippis I just missread you feedback

File: dev-python/pypeg2/pypeg2-2.15.2.ebuild [hunk 4]
```diff
@@ -20,6 +20,7 @@ SLOT="0"
 KEYWORDS="~amd64 ~x86"
 
 RDEPEND="dev-python/lxml[${PYTHON_USEDEP}]"
+BDEPEND="dev-python/setuptools[${PYTHON_USEDEP}]"
```
Comment (juippis) [inline dev-python/pypeg2/pypeg2-2.15.2.ebuild]: There is no BDEPEND with EAPI-5. It's 7+ thing. So only modify PYTHON_COMPAT in this ebuild and don't add / touch anything else.

File: www-client/qutebrowser/qutebrowser-1.8.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,88 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=5
```
Comment (juippis) [inline www-client/qutebrowser/qutebrowser-1.8.1.ebuild]: This is still EAPI-5 which has been deprecated already, so upgrade to 7 since it should work.

File: www-client/qutebrowser/qutebrowser-1.8.1.ebuild [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+	>=dev-python/pyyaml-3.12[${PYTHON_USEDEP},libyaml]
+"
+
+# Tests restricted as the deplist (misc/requirements/requirements-tests.txt)
+# isn't complete and X11 is required in order to start up qutebrowser.
+RESTRICT="test"
+
+python_compile_all() {
+	if [[ ${PV} == "9999" ]]; then
+		"${PYTHON}" scripts/asciidoc2html.py || die "Failed generating docs"
```
Comment (juippis) [inline www-client/qutebrowser/qutebrowser-1.8.1.ebuild]: This should probably be `EPYTHON`. Do you intend to update -9999 ebuild as well? You can do it in a separate PR yeah, but asking should these "9999" clauses be dropped or not...

File: dev-python/pypeg2/pypeg2-2.15.2.ebuild (hunk 4 above)
Comment (GuillaumeSeren) [reply dev-python/pypeg2/pypeg2-2.15.2.ebuild]: Ok

File: www-client/qutebrowser/qutebrowser-1.8.1.ebuild (hunk 5 above)
Comment (GuillaumeSeren) [reply www-client/qutebrowser/qutebrowser-1.8.1.ebuild]: Ok

File: www-client/qutebrowser/qutebrowser-1.8.1.ebuild (hunk 6 above)
Comment (GuillaumeSeren) [reply www-client/qutebrowser/qutebrowser-1.8.1.ebuild]: Ah yes I can upgrade the live ebuild in this PR and maybe cleanup old version ?
Each in seperate commits

**Comment by juippis** [comment]
Your sign-offs were a bit malformed. It looks like you use repoman to commit, so in the future you can automate sign-offs by adding the correct format to make.conf with `SIGNED_OFF_BY`. `man 5 make.conf`
